Package fetching has regressed on Windows

### Zig Version

0.12.0-dev.878+7abf9b3a83

### Steps to Reproduce and Observed Behavior

Regressed in https://github.com/ziglang/zig/pull/17392

Building `zls` on latest master:

```
> zig build -Doptimize=ReleaseSafe
C:\Users\Ryan\Programming\Zig\zls\zig-cache\o\8c6656817d7b9ba969d47a90c375e93f\dependencies.zig:3:70: error: invalid escape character: 'p'
        pub const build_root = "C:\\Users\\Ryan\\AppData\\Local\\zig\p\\1220363c7e27b2d3f39de6ff6e90f9537a0634199860fea237a55ddb1e1717f5d6a5";
                                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Users\Ryan\Programming\Zig\zls\zig-cache\o\8c6656817d7b9ba969d47a90c375e93f\dependencies.zig:8:70: error: invalid escape character: 'p'
        pub const build_root = "C:\\Users\\Ryan\\AppData\\Local\\zig\p\\122089a8247a693cad53beb161bde6c30f71376cd4298798d45b32740c3581405864";
                                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Users\Ryan\Programming\Zig\zls\zig-cache\o\8c6656817d7b9ba969d47a90c375e93f\dependencies.zig:13:70: error: invalid escape character: 'p'
        pub const build_root = "C:\\Users\\Ryan\\AppData\\Local\\zig\p\\1220bb12c9bfe291eed1afe6a2070c7c39918ab1979f24a281bba39dfb23f5bcd544";
                                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

With `0.12.0-dev.819+75b48ef50` (one build before https://github.com/ziglang/zig/pull/17392 was merged):

```
> "C:\Users\Ryan\Programming\Zig\zig-downloaded\zig-windows-x86_64-0.12.0-dev.819+75b48ef50\zig" build -Doptimize=ReleaseSafe

```
(success)

### Expected Behavior

Successful fetching and compilation

--- a/src/Package.zig
+++ b/src/Package.zig
@@ -124,7 +124,7 @@ pub const Path = struct {
             };
             if (self.root_dir.path) |p| {
                 try stringEscape(p, f, options, writer);
-                if (self.sub_path.len > 0) try writer.writeAll(fs.path.sep_str);
+                if (self.sub_path.len > 0) try stringEscape(fs.path.sep_str, f, options, writer);
             }
             if (self.sub_path.len > 0) {
                 try stringEscape(self.sub_path, f, options, writer);

